src/Scene/NodeProcess.js

```diff
@@ -324,51 +318,41 @@ function updateNodeElevation(quadtree, node, layersConfig, force) {
     // We don't care about layer status (isLayerFrozen) or limits (tileInsideLimit) because
     // we simply want to use ancestor's texture with a different pitch
     if (currentElevation == -1) {
-        for (let i = 0; i < elevationLayers.length; i++) {
-            let layer = elevationLayers[i];
-
-            if (currentElevation === -1) {
-                let a = findAncestorWithValidTextureForLayer(node, l_ELEVATION, layer);
-
-                if (a != null) {
-                    bestLayer = layer;
-                    ancestor = a;
-                    break;
-                }
-            }
-        }
+        ancestor = findAncestorWithValidTextureForLayer(node, l_ELEVATION);
     }
 
     // We don't have a texture to reuse. This can happen in two cases:
     //   * no ancestor texture to use
     //   * we already have 1 texture (so currentElevation >= 0)
     // Again, LayeredMaterial's 1 elevation texture limitation forces us to `break` as soon
     // as one layer can supply a texture for this node. So ordering of elevation layers is important.
-    if (bestLayer == null) {
-        for (let i = 0; i < elevationLayers.length; i++) {
-            let layer = elevationLayers[i];
+    // Ordering way of loop is important to find the best layer with tileInsideLimit
+    for (let i = elevationLayers.length - 1; i >= 0; i--) {
+        let layer = elevationLayers[i];
 
-            if (!layer.tileInsideLimit(node, layer)) {
-                continue;
-            }
+        if (!layer.tileInsideLimit(node, layer)) {
+            continue;
+        }
 
-            if (layersConfig.isLayerFrozen(layer.id) && !force) {
-                continue;
-            }
+        if (layersConfig.isLayerFrozen(layer.id) && !force) {
+            continue;
+        }
 
-            let targetLevel = chooseNextLevelToFetch(layer.updateStrategy.type, node.level, currentElevation, layer.updateStrategy.options);
-            if (targetLevel <= currentElevation) {
-                continue;
-            }
+        let targetLevel = chooseNextLevelToFetch(layer.updateStrategy.type, node.level, currentElevation, layer.updateStrategy.options);
+        if (targetLevel <= currentElevation) {
+            continue;
+        }
 
+        if(ancestor === null) {
```

Comment:
This change is weird and the behavior is not obvious.
I understand why you did this but this is acceptable only because it's meant to be temporary.

Add a comment explaining what is going on (either here or when you define `ancestor` line 321):
```js
// ancestor is not enough: we also need to know from which layer we're going to request the elevation texture (see how this is done for color texture).
// Right now this is done in the `for` loop below but this is hacky because there's no real warranty that bestLayer and ancestor really match.
// FIXME: we need to be able to set both ancestor and bestLayer at the same time